Logout with client id parameter

patryk-sob opened the pull request:
Logout with client_id parameter is allowed due to openid specification. When id_token_hint is not specified, client_id can be used instead. This change is a solution for situation when session_token is expired and in the logout operation we get the error message "Missing Parameters: id_token_hint error on Session timeout.".

patryk-sob committed:
Logout with client id

Logout with client_id parameter is allowed due to openid specification. When id_token_hint is not specified, client_id can be used instead. This change is a solution for situation when session_token is expired and in the logout operation we get the error message "Missing Parameters: id_token_hint error on Session timeout.".

patryk-sob committed:
Update openidc.lua

bodewig commented:
could you please add a new configuration param like `redirect_after_logout_with_id_token_hint` for this and document it? Bonus points for extending https://github.com/zmartzone/lua-resty-openidc/blob/master/tests/spec/logout_spec.lua ;-)

patryk-sob committed:
Merge remote-tracking branch 'upstream/master' into logout-with-client-id

patryk-sob committed:
Logout with client id parameter

- Implementation of logout with client_id parameter instead of id_token_hint parameter (openidc.lua)
- New test scenario for logout with client_id parameter (logout_spec.lua)
- Document new configuration parameter redirect_after_logout_with_client_id (README.md)
